File: programmers/92334.js
function solution(id_list, report, k) {
    const len = id_list.length
    const map = new Map()
    const rep = new Array(len).fill(0).map(() => new Array(len).fill(false))
    const time = new Array(len).fill(0)
    const answer = new Array(len).fill(0)
    
    let i = 0
    id_list.forEach(id => map.set(id, i++))
    
    report.forEach(log => {
        log = log.split(' ')
        
        const f = map.get(log[0])
        const t = map.get(log[1])
        
        if (!rep[t][f]) {
            time[t]++
            rep[t][f] = true
        }
    })
    
    for (i = 0; i < len; i++) if (time[i] >= k)
        for (let j = 0; j < len; j++) if (rep[i][j]) answer[j]++
    
    return answer
}

File: programmers/92334.java
package programmers;

import java.util.Arrays;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;
import java.util.stream.IntStream;

class Solution {
  public int[] solution(String[] id_list, String[] report, int k) {
    int[] answer = IntStream.range(0, id_list.length).toArray();
    Arrays.fill(answer, 0);

    Map<String, Integer> idMap = new HashMap<>();
    Map<Integer, Set<Integer>> reportMap = new HashMap<>();

    for (int i = 0; i < id_list.length; i++)
      idMap.put(id_list[i], i);

    for (String r : report) {
      String[] reportInfo = r.split(" ");
      Integer reporter = idMap.get(reportInfo[0]);
      Integer reportee = idMap.get(reportInfo[1]);

      if (reportMap.get(reportee) == null)
        reportMap.put(reportee, new HashSet<>());

      reportMap.get(reportee).add(reporter);
    }

    reportMap.values().stream().forEach((reportSet) -> {
      if (reportSet.size() < k)
        return;
      reportSet.stream().forEach((reporter) -> answer[reporter] += 1);
    });

    return answer;
  }
}